cmd/probe/main.go

```diff
@@ -128,11 +133,16 @@ func main() {
 	}
 
 	// ha dependent on dbmanager which is initialized by rt.Run
-	// ha := highavailability.NewHa(logHa)
-	// if ha != nil {
-	// 	defer ha.ShutdownWithWait()
-	// 	go ha.Start()
-	// }
+	characterType := viper.GetString(constant.KBEnvCharacterType)
+	workload := viper.GetString(constant.KBEnvWorkloadType)
+	if strings.EqualFold(characterType, "mongodb") ||
+		(strings.EqualFold(characterType, "mysql") && strings.EqualFold(workload, Replication)) {
```

Comment:
can you enum the characterType  and workload? then only check if it hits the enum.